packages/fluentui/react-northstar/src/components/Menu/MenuDivider.tsx

```diff
@@ -9,75 +21,105 @@ import {
   ChildrenComponentProps,
   ContentComponentProps,
   rtlTextContainer,
+  ShorthandFactory,
+  ShorthandConfig,
 } from '../../utils';
-import { WithAsProp, withSafeTypeForAs, FluentComponentStaticProps, ProviderContextPrepared } from '../../types';
-import { useTelemetry, getElementType, useUnhandledProps, useAccessibility, useStyles } from '@fluentui/react-bindings';
-// @ts-ignore
-import { ThemeContext } from 'react-fela';
+import { ProviderContextPrepared } from '../../types';
 
 export interface MenuDividerProps extends UIComponentProps, ChildrenComponentProps, ContentComponentProps {
   /** Accessibility behavior if overridden by the user. */
-  accessibility?: Accessibility<never>;
+  accessibility?: Accessibility<MenuDividerBehaviorProps>;
 
   vertical?: boolean;
   primary?: boolean;
   secondary?: boolean;
   inSubmenu?: boolean;
 }
 
-export const menuDividerClassName = 'ui-menu__divider';
-
 export type MenuDividerStylesProps = Required<Pick<MenuDividerProps, 'vertical' | 'inSubmenu' | 'primary'>> & {
   hasContent: boolean;
 };
 
-export const MenuDivider: React.FC<WithAsProp<MenuDividerProps>> &
-  FluentComponentStaticProps<MenuDividerProps> = props => {
-  const context: ProviderContextPrepared = React.useContext(ThemeContext);
-  const { setStart, setEnd } = useTelemetry(MenuDivider.displayName, context.telemetry);
-  setStart();
-  const { children, content, vertical, inSubmenu, primary, className, design, styles, variables } = props;
-  const ElementType = getElementType(props);
-  const unhandledProps = useUnhandledProps(MenuDivider.handledProps, props);
-
-  const getA11yProps = useAccessibility(props.accessibility, {
-    debugName: MenuDivider.displayName,
-    rtl: context.rtl,
-  });
-
-  const { classes } = useStyles<MenuDividerStylesProps>(MenuDivider.displayName, {
+export const menuDividerClassName = 'ui-menu__divider';
+
+/**
+ * A MenuDivider is non-actionable element that visually segments items of Menu.
+ */
+const MenuDivider = compose<'li', MenuDividerProps, MenuDividerStylesProps, {}, {}>(
+  (props, ref, composeOptions) => {
+    const context: ProviderContextPrepared = React.useContext(ThemeContext);
+    const { setStart, setEnd } = useTelemetry(composeOptions.displayName, context.telemetry);
+    setStart();
+
+    const { children, content, vertical, inSubmenu, primary, className, design, styles, variables } = props;
+
+    const getA11yProps = useAccessibility(props.accessibility, {
+      debugName: composeOptions.displayName,
+      rtl: context.rtl,
+    });
+
+    const { classes } = useStyles<MenuDividerStylesProps>(composeOptions.displayName, {
```

Comment:
Let's move the styles from `menuStyles.ts`'s divider slots to new file, and update `Menu` to send these props to the `MenuDivider`.

Reply:
Good call, please check the latest commit